quadro.c

```diff
@@ -1,52 +1,104 @@
 #include <stdio.h>
 #include <stdbool.h>
 #include <math.h>
+#include <stdlib.h>
 
-bool sign_of_diskr(double a, double b, double c){
-    double D = (b*b - 4*a*c);
-    if (D>0)
+struct coef
+{
+    double a;
+    double b;
+    double c;
+};
+
+struct roots
+{
+    double* root1;
+    double* root2;
+};
+
+struct coef quadro_input() {
+    struct coef coefs;
+    coefs.a = 0;
+    coefs.b = 0;
+    coefs.c = 0;
+    printf("Enter coefficient a: ");
+    while (scanf("%lf", &coefs.a) != 1) {
+        printf("Error input for a, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    printf("Enter coefficient b: ");
+    while (scanf("%lf", &coefs.b) != 1) {
+        printf("Error input for b, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    printf("Enter coefficient c: ");
+    while (scanf("%lf", &coefs.c) != 1) {
+        printf("Error input for c, try again: ");
+        while (getchar() != '\n'); // очистка буфера ввода
+    }
+
+    return coefs;
+}
+
+bool sign_of_diskr(double d){
+    if (d>0)
         return(true);
     else
         return(false);
 }
 
 double diskr(double a, double b, double c){
-    double D = (b*b - 4*a*c);
-    return (D);
+    return (b*b - 4*a*c);
+}
+
+struct roots quadro_output(double a, double b, double c) {
+    struct roots roots;
```

Comment:
we use c++ compiler for our projects. that's why we can drop `struct` keyword when we initialize some value. you can use it only when you define some new struct. also, change file extension from `.c` to `.cpp`